platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/geometry/LatLngQuad.java

```diff
@@ -0,0 +1,95 @@
+package com.mapbox.mapboxsdk.geometry;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.support.annotation.NonNull;
+
+import com.mapbox.mapboxsdk.exceptions.InvalidLatLngBoundsException;
+import com.mapbox.services.android.telemetry.constants.GeoConstants;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A geographical area representing a non-aligned rectangle
+ * <p>
+ * This class does not wrap valies to the world bounds
+ * </p>
+ */
+public class LatLngQuad implements Parcelable {
```

Comment:
Regarding the Objective-C struct’s name, it may be worth seeing if there’s precedent for a concept like this in SceneKit or SpriteKit and naming the struct accordingly. The Objective-C struct’s name doesn’t have to match the Java class’s name – after all, the Android SDK has `LatLng` and the iOS/macOS SDK has `CLLocationCoordinate2D`.